docs/tutorials/azure.md

```diff
@@ -0,0 +1,166 @@
+
+```
+$ az group create -n externaldns -l eastus
+```
+
+Substitute a more suitable location for the resource group if desired.
+
+Next, create a Azure DNS zone for "example.com":
+
+```
+$ az network dns zone create -g externaldns -n example.com
+```
+
+Substitute a domain you own for "example.com" if desired.
+
+If using your own domain that was registered with a third-party domain registrar, you should point your domain's
+name servers to the values in the `nameServers` field from the JSON data returned by the `az network dns zone create` command.
+Please consult your registrar's documentation on how to do that.
+
+## Creating Azure Credentials Secret
+
+When your Kubernetes cluster is created by Azure Container Services, a file named `/etc/kubernetes/azure.json` is created to store
+the Azure credentials for API access.  Kubernetes uses this file for the Azure cloud provider.
+
+For ExternalDNS to access the Azure API, it also needs access to this file.  However, we will be deploying ExternalDNS inside of
+the Kubernetes cluster so we will need to use a Kubernetes secret.
+
+To create the secret:
+
+```
+$ kubectl create secret generic azure-credentials-secret --from-file=/etc/kubernetes/azure.json
+```
+
+## Deploy ExternalDNS
+
+Create a deployment file called `externaldns.yaml` with the following contents:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: external-dns
+spec:
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      containers:
+      - name: external-dns
+        image: registry.opensource.zalan.do/teapot/external-dns:v0.4.0
+        args:
+        - --source=service
+        - --domain-filter=example.com. # (optional) limit to only example.com domains; change to match the zone created above.
+        - --provider=azure
+        - --azure-config-file=/etc/azure-credentials-secret
+        - --azure-resource-group externaldns # (optional) override resource group to tutorial resource group
+        volumeMounts:
+        - name: azure-credentials-secret
+          mountPath: /etc/azure-credentials-secret
+          readOnly: true
+      volumes:
+      - name: azure-credentials-secret
+        secret:
+          secretName: azure-credentials-secret
+```
+
+Create the deployment for ExternalDNS:
+
+```
+$ kubectl create -f externaldns.yaml
+```
+
+## Deploying an Nginx Service
+
+Create a service file called 'nginx.yaml' with the following contents:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: nginx
+spec:
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - image: nginx
+        name: nginx
+        ports:
+        - containerPort: 80
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx
+  annotations:
+    external-dns.alpha.kubernetes.io/hostname: example.com.
```

Comment:
shouldn't that be a specific record in the zone `example.com`, like `foo.example.com`?